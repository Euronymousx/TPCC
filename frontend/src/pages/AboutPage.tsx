import { motion } from 'framer-motion'

export function AboutPage() {
  return (
    <motion.section initial={{ opacity: 0, y: 40 }} animate={{ opacity: 1, y: 0 }} className="container mx-auto py-16 space-y-6">
      <h2 className="text-3xl font-display text-brandBrown">About Us</h2>
      <p>Founded with a passion for pets, TPCC has served our community for over a decade providing bespoke care solutions.</p>
      <p>Our experienced team treats every pet like family, ensuring comfort and safety at all times.</p>
    </motion.section>
  )
}
