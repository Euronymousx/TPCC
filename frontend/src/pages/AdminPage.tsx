import { useState } from 'react'
import { motion } from 'framer-motion'
import { useAuth } from '../context/AuthProvider'
import { Button } from '../components/ui/Button'
import { v4 as uuidv4 } from 'uuid'

export function AdminPage() {
  const { applications, login } = useAuth()
  const [unlocked, setUnlocked] = useState(false)
  const [pass, setPass] = useState('')

  if (!unlocked) {
    const tryUnlock = () => {
      if (pass === 'AdminPass!23') {
        login('admin@tpcc.local', 'AdminPass!23')
        setUnlocked(true)
      }
    }
    return (
      <motion.div initial={{ opacity: 0 }} animate={{ opacity: 1 }} className="container mx-auto py-16 max-w-sm space-y-4">
        <input type="password" placeholder="Password" value={pass} onChange={(e) => setPass(e.target.value)} className="border px-3 py-2 w-full" />
        <Button onClick={tryUnlock}>Enter</Button>
      </motion.div>
    )
  }

  const copy = (id: string) => navigator.clipboard.writeText(`https://tpcc.local/register/temp/${id}`)

  return (
    <section className="container mx-auto py-16 space-y-4">
      <h2 className="text-3xl font-display">Applications</h2>
      <ul className="space-y-2">
        {applications.map((a, i) => (
          <li key={i} className="bg-offWhite rounded-card p-4 flex justify-between items-center">
            <span>{(a as any).client?.name || (a as any).name}</span>
            <Button onClick={() => copy(uuidv4())}>Generate Temporary Signup Link</Button>
          </li>
        ))}
      </ul>
    </section>
  )
}
