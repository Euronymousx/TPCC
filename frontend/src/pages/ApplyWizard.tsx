import { useState } from 'react'
import { useForm, useFieldArray } from 'react-hook-form'
import { z } from 'zod'
import { zodResolver } from '@hookform/resolvers/zod'
import axios from 'axios'
import { Button } from '../components/ui/Button'
import { Input } from '../components/ui/Input'
import { useAuth } from '../context/AuthProvider'
import { motion } from 'framer-motion'

const petSchema = z.object({
  name: z.string().min(1),
  breed: z.string().min(1),
  age: z.string().min(1),
})

const formSchema = z.object({
  client: z.object({
    name: z.string().min(1),
    email: z.string().email(),
    phone: z.string().min(1),
  }),
  pets: z.array(petSchema),
  services: z.array(z.string()),
})

type FormData = z.infer<typeof formSchema>

export function ApplyWizard() {
  const [step, setStep] = useState(1)
  const { addApplication } = useAuth()

  const form = useForm<FormData>({
    resolver: zodResolver(formSchema),
    defaultValues: { pets: [{ name: '', breed: '', age: '' }], services: [] },
  })
  const { register, control, handleSubmit, watch } = form
  const { fields, append, remove } = useFieldArray({ control, name: 'pets' })

  const next = handleSubmit(() => setStep((s) => s + 1))
  const prev = () => setStep((s) => s - 1)

  const onSubmit = handleSubmit(async (data) => {
    await axios.post('/api/applications', data)
    addApplication(data)
    setStep(5)
  })

  return (
    <motion.section initial={{ opacity: 0 }} animate={{ opacity: 1 }} className="container mx-auto py-16 max-w-xl space-y-4">
      <div className="mb-4 h-2 bg-brandBrown/20 rounded">
        <div className="bg-brandBrown h-full" style={{ width: `${(step - 1) / 4 * 100}%` }} />
      </div>
      {step === 1 && (
        <form onSubmit={next} className="space-y-4">
          <Input placeholder="Full Name" {...register('client.name')} />
          <Input placeholder="Email" type="email" {...register('client.email')} />
          <Input placeholder="Phone" {...register('client.phone')} />
          <Button type="submit">Next</Button>
        </form>
      )}
      {step === 2 && (
        <form onSubmit={next} className="space-y-4">
          {fields.map((field, idx) => (
            <div key={field.id} className="space-y-2 border p-3 rounded-xl">
              <Input placeholder="Pet Name" {...register(`pets.${idx}.name` as const)} />
              <Input placeholder="Breed" {...register(`pets.${idx}.breed` as const)} />
              <Input placeholder="Age" {...register(`pets.${idx}.age` as const)} />
              {fields.length > 1 && (
                <Button type="button" variant="secondary" onClick={() => remove(idx)}>Remove</Button>
              )}
            </div>
          ))}
          <Button type="button" variant="secondary" onClick={() => append({ name: '', breed: '', age: '' })}>Add Pet</Button>
          <div className="flex gap-2">
            <Button type="button" variant="secondary" onClick={prev}>Back</Button>
            <Button type="submit">Next</Button>
          </div>
        </form>
      )}
      {step === 3 && (
        <form onSubmit={next} className="space-y-4">
          <label className="block"><input type="checkbox" value="Daily walks" {...register('services')} /> Daily walks</label>
          <label className="block"><input type="checkbox" value="Drop-in visits" {...register('services')} /> Drop-in visits</label>
          <label className="block"><input type="checkbox" value="Pet taxi" {...register('services')} /> Pet taxi</label>
          <label className="block"><input type="checkbox" value="Grooming scheduling" {...register('services')} /> Grooming scheduling</label>
          <div className="flex gap-2">
            <Button type="button" variant="secondary" onClick={prev}>Back</Button>
            <Button type="submit">Review</Button>
          </div>
        </form>
      )}
      {step === 4 && (
        <form onSubmit={next} className="space-y-4">
          <pre className="bg-offWhite p-4 rounded-xl text-sm overflow-auto max-h-52">{JSON.stringify(watch(), null, 2)}</pre>
          <div className="flex gap-2">
            <Button type="button" variant="secondary" onClick={prev}>Back</Button>
            <Button type="submit">Submit</Button>
          </div>
        </form>
      )}
      {step === 5 && (
        <p className="text-center">Weâ€™ll contact you to schedule a meet-and-greet before account setup.</p>
      )}
    </motion.section>
  )
}
