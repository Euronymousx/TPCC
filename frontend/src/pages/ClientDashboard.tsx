export function ClientDashboard() {
  return (
    <div className="container mx-auto py-16">
      <h2 className="text-3xl font-display text-brandBrown mb-4">Client Dashboard</h2>
      <p>Welcome to your dashboard.</p>
    </div>
  )
}
