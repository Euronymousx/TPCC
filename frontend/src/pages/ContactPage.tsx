import { useForm } from 'react-hook-form'
import { zodResolver } from '@hookform/resolvers/zod'
import { z } from 'zod'
import { Input } from '../components/ui/Input'
import { Textarea } from '../components/ui/Textarea'
import { Button } from '../components/ui/Button'
import { motion } from 'framer-motion'

const schema = z.object({
  name: z.string().min(1, 'Required'),
  email: z.string().email('Invalid email'),
  message: z.string().min(1, 'Required'),
})

export function ContactPage() {
  const { register, handleSubmit, formState: { errors } } = useForm({ resolver: zodResolver(schema) })

  const onSubmit = handleSubmit((data) => {
    console.log(data)
  })

  return (
    <motion.section id="contact" initial={{ opacity: 0, y: 40 }} animate={{ opacity: 1, y: 0 }} className="container mx-auto py-16 max-w-lg space-y-4">
      <h2 className="text-3xl font-display text-brandBrown text-center">Contact Us</h2>
      <form onSubmit={onSubmit} className="space-y-4">
        <div>
          <Input placeholder="Name" {...register('name')} />
          {errors.name && <p className="text-red-500 text-sm">{errors.name.message as string}</p>}
        </div>
        <div>
          <Input placeholder="Email" type="email" {...register('email')} />
          {errors.email && <p className="text-red-500 text-sm">{errors.email.message as string}</p>}
        </div>
        <div>
          <Textarea placeholder="Message" rows={4} {...register('message')} />
          {errors.message && <p className="text-red-500 text-sm">{errors.message.message as string}</p>}
        </div>
        <Button type="submit">Send Message</Button>
      </form>
      <div className="h-64 bg-brandBrown/10 rounded-card flex items-center justify-center">
        Map placeholder
      </div>
    </motion.section>
  )
}
