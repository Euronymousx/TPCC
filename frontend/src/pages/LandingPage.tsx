import { Hero } from '../components/Hero'
import { motion } from 'framer-motion'

export function LandingPage() {
  return (
    <div>
      <Hero />
      <motion.section initial={{ opacity: 0, y: 40 }} whileInView={{ opacity: 1, y: 0 }} viewport={{ once: true }} className="container mx-auto py-16 space-y-8">
        <h2 className="text-3xl font-display text-center text-brandBrown">Our Promise</h2>
        <p className="text-center max-w-2xl mx-auto">We deliver unmatched pet care and concierge services so you can enjoy peace of mind knowing your companions are in expert hands.</p>
      </motion.section>
    </div>
  )
}
