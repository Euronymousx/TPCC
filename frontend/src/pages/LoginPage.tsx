import { useForm } from 'react-hook-form'
import { Input } from '../components/ui/Input'
import { Button } from '../components/ui/Button'
import { useAuth } from '../context/AuthProvider'
import { useNavigate } from 'react-router-dom'
import { motion } from 'framer-motion'

export function LoginPage() {
  const { login } = useAuth()
  const navigate = useNavigate()
  const { register, handleSubmit } = useForm()

  const onSubmit = handleSubmit((data) => {
    login(data.email, data.pass)
    navigate('/dashboard')
  })

  return (
    <motion.form initial={{ opacity: 0 }} animate={{ opacity: 1 }} onSubmit={onSubmit} className="container mx-auto py-16 max-w-sm space-y-4">
      <Input placeholder="Email" type="email" {...register('email')} />
      <Input placeholder="Password" type="password" {...register('pass')} />
      <Button type="submit">Login</Button>
    </motion.form>
  )
}
