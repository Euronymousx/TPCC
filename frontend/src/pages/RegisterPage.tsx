import { useForm } from 'react-hook-form'
import { Input } from '../components/ui/Input'
import { Button } from '../components/ui/Button'
import { useNavigate } from 'react-router-dom'
import { motion } from 'framer-motion'

export function RegisterPage() {
  const navigate = useNavigate()
  const { register, handleSubmit } = useForm()

  const onSubmit = handleSubmit(() => {
    navigate('/login')
  })

  return (
    <motion.form initial={{ opacity: 0 }} animate={{ opacity: 1 }} onSubmit={onSubmit} className="container mx-auto py-16 max-w-sm space-y-4">
      <Input placeholder="Username" {...register('user')} />
      <Input placeholder="Email" {...register('email')} />
      <Input placeholder="Password" type="password" {...register('pass')} />
      <Button type="submit">Register</Button>
    </motion.form>
  )
}
