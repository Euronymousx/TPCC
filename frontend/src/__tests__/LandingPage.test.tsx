import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import { BrowserRouter } from 'react-router-dom'
import { LandingPage } from '../pages/LandingPage'

describe('LandingPage', () => {
  it('renders logo and CTA', () => {
    const { container } = render(
      <BrowserRouter>
        <LandingPage />
      </BrowserRouter>
    )
    expect(screen.getByAltText('TPCC logo')).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /apply online/i })).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
