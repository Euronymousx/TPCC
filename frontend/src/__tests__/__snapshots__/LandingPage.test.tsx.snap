// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LandingPage > renders logo and CTA 1`] = `
<div>
  <div>
    <section
      class="relative h-screen flex flex-col items-center justify-center bg-offWhite text-nearBlack"
    >
      <img
        alt="TPCC logo"
        class="w-40 md:w-60 mb-8"
        src="/img/logo-walker.png"
        style="opacity: 0; transform: translateY(-20px);"
      />
      <h1
        class="text-center text-4xl md:text-6xl font-display"
        style="opacity: 0; transform: translateY(20px);"
      >
        Luxury Dog-Walking & Pet Care, Reimagined.
      </h1>
      <p
        class="mt-4 text-center max-w-xl"
        style="opacity: 0; transform: translateY(20px);"
      >
        Daily walks, safety-first practices, concierge-level service.
      </p>
      <div
        class="mt-8 flex gap-4"
        style="opacity: 0;"
      >
        <button
          class="transition-transform shadow-xl rounded-card px-5 py-3 font-semibold focus:outline-none bg-brandBrown text-offWhite hover:-translate-y-1"
        >
          Apply Online
        </button>
        <button
          class="transition-transform shadow-xl rounded-card px-5 py-3 font-semibold focus:outline-none bg-accentTeal text-nearBlack hover:-translate-y-1"
        >
          See Pricing
        </button>
      </div>
    </section>
    <ul
      class="bg-offWhite py-8 flex flex-wrap justify-center gap-4 text-sm"
      style="opacity: 0;"
    >
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        Background-checked
      </li>
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        High-vis gear
      </li>
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        Belt-attached leashes
      </li>
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        CPR certified
      </li>
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        Insured & bonded
      </li>
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        Crash-tested transport
      </li>
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        Body-cams
      </li>
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        Two-walker teams
      </li>
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        GPS trackers
      </li>
      <li
        class="px-3 py-1 bg-brandBrown text-offWhite rounded-full shadow"
      >
        Heated winter jackets
      </li>
    </ul>
    <section
      class="container mx-auto py-16 space-y-10"
      id="pricing"
    >
      <h2
        class="text-3xl font-display text-center text-brandBrown"
        style="opacity: 0; transform: translateY(20px);"
      >
        Subscription Pricing
      </h2>
      <div
        class="grid md:grid-cols-2 gap-8"
      >
        <div
          class="bg-offWhite shadow-xl rounded-card p-6 text-center space-y-4"
        >
          <h3
            class="text-2xl font-display text-brandBrown"
          >
            Base Subscription
          </h3>
          <p
            class="text-4xl font-display"
          >
            $
            750
             
            <span
              class="text-lg font-sans"
            >
              / month
            </span>
          </p>
          <p>
            1 daily walk • 5 drop-ins • eligible for add-ons
          </p>
        </div>
        <div
          class="bg-offWhite shadow-xl rounded-card p-6 space-y-4"
        >
          <h4
            class="font-display text-xl text-brandBrown"
          >
            Add-ons
          </h4>
          <label
            class="flex items-center gap-2"
          >
            <input
              type="checkbox"
            />
            Another dog +30% ($
            225.00
            )
          </label>
          <label
            class="flex items-center gap-2"
          >
            <input
              type="checkbox"
            />
            2nd walk / day +75% ($
            562.50
            )
          </label>
          <p
            class="font-semibold"
          >
            Total: $
            750.00
          </p>
          <div>
            <p>
              Extra drop-ins $60 ea | 5-pack $275 | 10-pack $500
            </p>
            <p>
              Weekly bath $40 / visit | Bi-weekly $55
            </p>
            <p>
              Grooming scheduling (price TBD, 
              <a
                class="underline"
                href="#"
              >
                Ask us
              </a>
              )
            </p>
          </div>
        </div>
      </div>
      <div
        class="bg-offWhite shadow-xl rounded-card p-6 space-y-4 max-w-md mx-auto"
      >
        <h4
          class="font-display text-xl text-brandBrown"
        >
          Pet-Taxi Calculator
        </h4>
        <div
          class="space-y-2"
        >
          <label
            class="block text-sm"
          >
            Miles:
            <input
              class="border px-2 py-1 ml-2 w-20"
              type="number"
              value="0"
            />
          </label>
          <label
            class="block text-sm"
          >
            Waiting hours:
            <input
              class="border px-2 py-1 ml-2 w-20"
              type="number"
              value="0"
            />
          </label>
          <p
            class="font-semibold"
          >
            $
            40.00
          </p>
        </div>
      </div>
    </section>
    <section
      class="container mx-auto py-16 grid md:grid-cols-3 gap-8"
    >
      <figure
        class="shadow-2xl rounded-3xl overflow-hidden"
      >
        <img
          alt="Happy dog on walk"
          class="w-full"
          src="https://via.placeholder.com/600x400"
        />
        <figcaption
          class="p-2 text-center text-sm"
        >
          Happy dog on walk
        </figcaption>
      </figure>
      <figure
        class="shadow-2xl rounded-3xl overflow-hidden"
      >
        <img
          alt="Two walkers in high-vis gear"
          class="w-full"
          src="https://via.placeholder.com/600x400"
        />
        <figcaption
          class="p-2 text-center text-sm"
        >
          Two walkers in high-vis gear
        </figcaption>
      </figure>
      <figure
        class="shadow-2xl rounded-3xl overflow-hidden"
      >
        <img
          alt="Crash-tested kennel in van"
          class="w-full"
          src="https://via.placeholder.com/600x400"
        />
        <figcaption
          class="p-2 text-center text-sm"
        >
          Crash-tested kennel in van
        </figcaption>
      </figure>
    </section>
  </div>
</div>
`;
